examples/fe-tools/gensx/agent.ts

```diff
@@ -0,0 +1,236 @@
+                    ...contentParts,
+                  ];
+                  publishMessages();
+                } else if (chunk.type === "reasoning") {
+                  accumulatedReasoning += chunk.textDelta;
+
+                  // Update or add reasoning part
+                  const existingReasoningPartIndex = contentParts.findIndex(
+                    (part) => part.type === "reasoning",
+                  );
+                  if (existingReasoningPartIndex >= 0) {
+                    const reasoningPart =
+                      contentParts[existingReasoningPartIndex];
+                    if (reasoningPart.type === "reasoning") {
+                      reasoningPart.text = accumulatedReasoning;
+                    }
+                  } else {
+                    contentParts.push({
+                      type: "reasoning",
+                      text: accumulatedReasoning,
+                    });
+                  }
+
+                  allMessages[assistantMessageIndex].content = [
+                    ...contentParts,
+                  ];
+                  publishMessages();
+                } else if (chunk.type === "tool-call") {
+                  // Add tool call part - ensure args is an object, not a string
+                  let parsedArgs = chunk.args;
+                  if (typeof chunk.args === "string") {
+                    try {
+                      parsedArgs = JSON.parse(chunk.args);
+                    } catch {
+                      console.warn("Failed to parse tool args:", chunk.args);
+                      parsedArgs = chunk.args;
+                    }
+                  }
+
+                  contentParts.push({
+                    type: "tool-call",
+                    toolCallId: chunk.toolCallId,
+                    toolName: chunk.toolName,
+                    args: parsedArgs,
+                  });
+
+                  allMessages[assistantMessageIndex].content = [
+                    ...contentParts,
+                  ];
+                  publishMessages();
+                }
+
+                controller.enqueue(chunk);
+              },
+              flush() {
+                // Final publish when stream is complete
+                publishMessages();
+              },
+            });
+
+            return {
+              stream: stream.pipeThrough(transformStream),
+              ...rest,
+            };
+          },
+        },
+      ],
+    });
+
+    // Add system message for geographic focus
+    const systemMessage: CoreMessage = {
+      role: "system",
+      content: `You are a helpful geographic assistant that can interact with an interactive map. You have access to several map tools:
+
+- webSearch: Search the web for information relevant to the user's query
+- geocode: Geocode a location from an address or a query to a specific location, returned with latitude and longitude, as well as other useful information about the location
```

Comment:
I'd probably define this in `workflows.ts` instead so we can easily rip out this code once we have the agent abstraction.

In `loadChatHistory()` you can add the system message to the messages if history is undefined: `return history ?? [systemMessage];`